clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5099,12 +5099,17 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   if (Args.hasArg(options::OPT_fclangir_mem2reg))
     CmdArgs.push_back("-fclangir-mem2reg");
 
+  if (Args.hasArg(options::OPT_fclangir_idiom_recognizer,
+                  options::OPT_fclangir_idiom_recognizer_EQ))
+    CmdArgs.push_back("-fclangir-idiom-recognizer");
+
   // ClangIR lib opt requires idiom recognizer.
   if (Args.hasArg(options::OPT_fclangir_lib_opt,
                   options::OPT_fclangir_lib_opt_EQ)) {
     if (!Args.hasArg(options::OPT_fclangir_idiom_recognizer,
```

Comment:
While here (and since you are using it above) perhaps refactor this check into a boolean variable that can be used by both